### PR TITLE
Relax check during DROP PART

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix rare bug with `DROP PART` query for `ReplicatedMergeTree` tables which can lead to error message `Unexpected merged part intersecting drop range`.

### CODE REVIEW

**Comment by alesapin** [comment]
@mergifyio update

**Comment by alesapin** [comment]
@Mergifyio update

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -2332,10 +2332,33 @@ MergeTreeData::DataPartsVector MergeTreeData::removePartsInRangeFromWorkingSet(c
             throw Exception("Unexpected partition_id of part " + part->name + ". This is a bug.", ErrorCodes::LOGICAL_ERROR);
 
         /// It's a DROP PART and it's already executed by fetching some covering part
-        if (part->info != drop_range && part->info.contains(drop_range))
+        bool is_drop_part = !drop_range.isFakeDropRangePart();
```
Comment (tavplubix) [inline src/Storages/MergeTree/MergeTreeData.cpp:2335]: It also can be `drop partition`. But drop range for partition always have zero `min_block`. Drop range for part also may have zero `min_block`, but such drop range cannot be intersected my merge entry in the way described in the comment. So we can relax check only for some drop ranges for parts, not for partitions.
```suggestion
        bool is_drop_part = !drop_range.isFakeDropRangePart() && drop_range.min_block;
```

**Review by tavplubix — APPROVED** [review]

**Comment by alesapin** [comment]
No related failures.

**Comment by alesapin** [comment]
However, I'll fix this in ReplicatedQueue. But this check is also useful.

**Pull request merged**